### DISCUSSION TITLE
Can I generate TOC items based on .md files in directory?

### DISCUSSION THREAD

**Discussion by danroot** [dotnet/docfx#8850] [Q&A] [answered]
My end goal is to let documentation writers only have to place a .md file in the project without needing to update toc.yml when adding documentation.  I'm not set on any particular way of accomplishing that, but am envisioning:

toc.yml
```
- name: How-to Documents
   href: howtos/index.md
   items: howtos/**.md
```

/howtos/foo.md
```
---
name: How to Foo
---
article on how to foo
```

/howtos/bar.md
```
---
name: How to Bar
---
article on how to bar
```

Resulting TOC:
```
How to Documents (links to index.md)
|_ How to Foo (links to howtos/foo.md)
|_ How to Bar (links to howtos/bar.md)
```

Is this possible today out-of-the-box?  If not, what would be the "easiest" way to go about adding this?  I've thought about writing some powershell script to generate the toc.yml, but wasn't sure that was best approach.

**Comment by filzrev** [comment, marked as answer]
Currently there is no official tool to creating `toc.yml`  based on folder structure. (See: #3124).
But some third-party tools are available for this purpose. 

For example, following pages show [DocFxTocGenerator](https://github.com/Ellerbach/docfx-companion-tools) tool.

> https://dotnet.github.io/docfx/extensions/tools.html

#### Usage of DocFxTocGenerator 

```
dotnet tool install DocFxTocGenerator --global
DocFxTocGenerator -d ./docs
```
